nwbinspector/checks/ecephys.py

```diff
@@ -79,3 +81,17 @@ def check_spike_times_not_in_unobserved_interval(units_table: Units, nunits: int
                     "observed intervals."
                 )
             )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=ElectrodeTable)
+def check_optional_columns_electrode_table(electrode_table: ElectrodeTable):
+    """Check that the ElectrodeTable columns which are optional for nwb-schema>=2.5.0 are not specified as NaN."""
+    if get_package_version(name="pynwb") < version.Version("2.1.0"):
+        return
+    if any(x.isnan() in electrode_table for x in ["x", "y", "z", "imp", "filtering"]):
+        return InspectorMessage(
```

Comment:
I would yield here and say in the message which column is Nan. Also, does `isnan` work for an array? Maybe we could say `if np.all(np.isnan(col))`

Reply:
Sure, do you want it to report the exact index(ices) that are `NaN` or just the column that it was found in?